// stimulus-sortable/dist/stimulus-sortable.mjs
import { Controller as n } from "@hotwired/stimulus";
import s from "sortablejs";
import { patch as o } from "@rails/request.js";
class r extends n {
  initialize() {
    this.onUpdate = this.onUpdate.bind(this);
  }
  connect() {
    this.sortable = new s(this.element, {
      ...this.defaultOptions,
      ...this.options
    });
  }
  disconnect() {
    this.sortable.destroy(), this.sortable = void 0;
  }
  async onUpdate({ item: t, newIndex: a }) {
    if (!t.dataset.sortableUpdateUrl)
      return;
    const i = this.resourceNameValue ? `${this.resourceNameValue}[${this.paramNameValue}]` : this.paramNameValue, e = new FormData();
    return e.append(i, a + 1), await o(t.dataset.sortableUpdateUrl, { body: e, responseKind: this.responseKindValue });
  }
  get options() {
    return {
      animation: this.animationValue || this.defaultOptions.animation || 150,
      handle: this.handleValue || this.defaultOptions.handle || void 0,
      onUpdate: this.onUpdate
    };
  }
  get defaultOptions() {
    return {};
  }
}
r.values = {
  resourceName: String,
  paramName: {
    type: String,
    default: "position"
  },
  responseKind: {
    type: String,
    default: "html"
  },
  animation: Number,
  handle: String
};
export {
  r as default
};
